### PR TITLE
FreeRTOS CMake file update

### PR DESCRIPTION
**Pull request by rmspacefish**
1. Added more warnings if paths are not found
2. Added new path suffixes because of changed FreeRTOS kernel structure

### CODE REVIEW

**Comment by g-arjones** [comment]
Unfortunately it looks like this is no longer being maintained. I have a few more changes as well... If you are willing to maintain a fork maybe we could merge it all...

**Comment by rmspacefish** [comment]
I'm not sure whether it is not maintained anymore or it just takes a long time.. In any case, the fastest solution probably would be to have a fork for now.. I already forked the project for my personal account, but I could fork the project for my own organization (which allows to me add people as well), merge your PRs and then merge my own PRs.  What do you think?

**Comment by g-arjones** [comment]
Sounds good to me...

**Comment by atsju** [comment]
It's maintained but slow. I already discussed with @Hish15 and we would like to help maintaining the repo to get PR merged faster. But we never took time to write to @ObKo to ask if is possible and how we could contribute. We will definitely do it but it will not be immediately.

**Comment by rmspacefish** [comment]
Sounds good, that would be nice. If the repository is transferred to an organization (free one is okay as well), you can also assign more people to be able to merge pull requests. If all PRs I require are merged, I will definitely switch back to the￼ upstream again

File: cmake/FindFreeRTOS.cmake [hunk 1]
```diff
@@ -15,9 +15,14 @@ endif()
 find_path(FreeRTOS_COMMON_INCLUDE
     NAMES FreeRTOS.h
     PATHS "${FREERTOS_PATH}" "${FREERTOS_PATH}/FreeRTOS" 
-    PATH_SUFFIXES  "Source/include"
+    PATH_SUFFIXES  "Source/include" "include"
     NO_DEFAULT_PATH
 )
+
+if(FreeRTOS_COMMON_INCLUDE MATCHES "FreeRTOS_COMMON_INCLUDE-NOTFOUND")
```
Comment (atsju) [inline cmake/FindFreeRTOS.cmake:22]:
```suggestion
if(NOT FreeRTOS_COMMON_INCLUDE)
```

File: cmake/FindFreeRTOS.cmake [hunk 2]
```diff
@@ -72,9 +77,18 @@ foreach(PORT ${FreeRTOS_FIND_COMPONENTS})
...
-        PATH_SUFFIXES "Source/portable/GCC/${PORT}"  "Source/portable/GCC/${PORT}/r0p1"
+        PATH_SUFFIXES
+            "portable/GCC/${PORT}/r0p1"
+            "portable/GCC/${PORT}"
+            "Source/portable/GCC/${PORT}"
+            "Source/portable/GCC/${PORT}/r0p1"
         NO_DEFAULT_PATH
     )
+
+    if(FreeRTOS_${PORT}_PATH MATCHES "FreeRTOS_${PORT}_PATH-NOTFOUND")
```
Comment (atsju) [inline cmake/FindFreeRTOS.cmake:88]:
```suggestion
    if(NOT FreeRTOS_${PORT}_PATH)
```

**Review by atsju — COMMENTED** [review]
Minor format remark. Appart from that it seems OK.

**Review by atsju — APPROVED** [review]

**Review by Hish15 — APPROVED** [review]

**Pull request merged**